### PR TITLE
PERF: unstack

### PR DESCRIPTION
**Pull request by jbrockmendel**
```
from asv_bench.benchmarks.reshape import *
self = ReshapeExtensionDtype()
self.setup("Period[s]")

%timeit self.time_unstack_fast("Period[s]")
23.2 ms ± 1.15 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)   # <- master
9.37 ms ± 128 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)  # <- PR
```

### CODE REVIEW

File: asv_bench/benchmarks/reshape.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def setup(self, dtype):
             values = np.take(list(string.ascii_letters), indices)
             values = [pd.Categorical(v) for v in values.T]
 
-        self.df = DataFrame(values, index, columns)
+        self.df = DataFrame({i: cat for i, cat in enumerate(values)}, index, columns)
```
Comment (jreback) [inline asv_bench/benchmarks/reshape.py:114]: why is this needed? (only for int)? e.g. category seems to do this already

Comment (jbrockmendel) [reply asv_bench/benchmarks/reshape.py:114]: ATM the constructor special-cases list-of-categorical, but that behavior is deprecated.  this is what we will want post-deprecation, and avoids a warning in the meantime

**Comment by jbrockmendel** [comment]
this may have broken one of the asvs, will check
